src/components/composer/composer_focus_store.ts

```diff
@@ -1,48 +1,62 @@
 import { SpreadsheetStore } from "../../stores/spreadsheet_store";
-import { ComposerSelection, ComposerStore } from "./composer/composer_store";
+import { ComposerSelection, EditionMode } from "./composer/cell_composer_store";
 
 export type ComposerFocusType = "inactive" | "cellFocus" | "contentFocus";
 
+export interface ComposerInterface {
+  id: string; // for testing purposes only
+  getEditionMode(): EditionMode;
+  startEdition(content?: string, selection?: ComposerSelection): void;
+  stopEdition(): void;
+  setCurrentContent(content: string, selection?: ComposerSelection): void;
+}
+
+interface Args {
+  focusMode?: ComposerFocusType;
+  content?: string;
+  selection?: ComposerSelection;
+}
+
 export class ComposerFocusStore extends SpreadsheetStore {
-  mutators = ["focusTopBarComposer", "focusGridComposerContent", "focusGridComposerCell"] as const;
-  private composerStore = this.get(ComposerStore);
+  mutators = ["focusComposer", "focusActiveComposer"] as const;
 
-  private topBarFocus: Exclude<ComposerFocusType, "cellFocus"> = "inactive";
-  private gridFocusMode: ComposerFocusType = "inactive";
+  private _activeComposer: ComposerInterface | undefined;
```

Comment:
we could have something like this:
```suggestion
  private activeComposer: ComposerInterface = VOID_COMPOSER;
```
with `VOID_COMPOSER` being something like

```js
const VOID_COMPOSER = {
  id: "void-composer",
  getEditionMode: () => "inactive",
  startEdition: () => {},
  stopEdition: () => {},
  setCurrentContent: () => {},
}
```
it would save a few ifs, throws and the `get activeComposer`.
What do you think ?

Reply:
the VOID_COMPOSER could throw, yes